### DISCUSSION TITLE
How can I ignore the timeout exception

### DISCUSSION THREAD

**Discussion by phuclh** [stefanzweifel/sidecar-browsershot#19] [Q&A] [answered]
Hi Stefan Zweifel,

Thank you for the great package!

Is there any way to ignore the timeout exception from Browershot? I cannot ignore the `LambdaExecutionException` from sidecar since it will ignore other exceptions.

**Comment by stefanzweifel** [comment, marked as answer]
Hey

The problem is that there is no dedicated `TimeoutException` in Browsershot. What you could do is this:

```php
use Hammerstone\Sidecar\Exceptions\LambdaExecutionException;
use Illuminate\Support\Str;
use Spatie\Browsershot\Browsershot;
use Wnx\SidecarBrowsershot\BrowsershotLambda;

try {
    $screenshot = BrowsershotLambda::url('https://github.com')
        ->base64Screenshot();
} catch (LambdaExecutionException $exception) {

    // Check if the Exception Message contains "TimeoutError".
    // If so, fallback to the local Browsershot instance.
    // Or you can do something else.
    if (Str::contains($exception->getMessage(), 'TimeoutError: Navigation timeout of')) {
        $screenshot = Browsershot::url('https://github.com')
            ->base64Screenshot();
    } else {
        // Rethrow the exception to be handled by the default Laravel Exception Handler
        throw $exception;
    }
}
```

By checking the message of the LambdaExecutionException you can fine tune your exception handling. If it isn't a TimeoutError, just rethrow the exception and let Laravel handle the exception like every other exception.

**Reply by phuclh** [reply]
@stefanzweifel Thank you 👯
